src/AutorestSwift/View Models/ServiceClientFileViewModel.swift

```diff
@@ -62,10 +63,20 @@ struct ServiceClientFileViewModel {
         self.paging = model.pagingNames
         self.protocols = paging != nil ? "PipelineClient, PageableClient" : "PipelineClient"
         var globalParameters = [ParameterViewModel]()
-        for globalParameter in model.globalParameters ?? [] where !globalParameter.name.starts(with: "$") {
-            globalParameters.append(ParameterViewModel(from: globalParameter))
+        var host: String?
+        for globalParameter in model.globalParameters ?? [] {
+            if globalParameter.name == "$host" {
+                host = globalParameter.value.clientDefaultValue
+            } else {
+                globalParameters.append(ParameterViewModel(from: globalParameter))
+            }
         }
         self.globalParameters = globalParameters
+        if let hostValue = host {
+            self.host = "\"\(hostValue)\""
+        } else {
+            self.host = ""
+        }
```

Comment:
I would suggest extracting this logic out to a private method since it's non-trivial. You can make it return a named tuple so you could have something like:

```swift
let globals = resolveGlobalParameters()
self.host = globals.host
self.globalParameters = globals.parameters 
```